python/mxnet/test_utils.py

```diff
@@ -1915,8 +1915,8 @@ def chi_square_check(generator, buckets, probs, nsamples=1000000):
     if continuous_dist:
         sample_bucket_ids = sample_bucket_ids // 2
     obs_freq = np.zeros(shape=len(buckets), dtype=np.int)
-    for i in range(len(buckets)):
-        obs_freq[i] = (sample_bucket_ids == i).sum()
+    for i, _ in enumerate(buckets):
+        obs_freq[i] = (sample_bucket_ids == buckets[i]).astype('uint8').sum()
```

Comment:
Are you sure this work? (sample_bucket_ids == buckets[i]) will return a bool object. you might hit AttributeError here. Please double check.

Reply:
hence I tried to convert it to int so as to take sum

Reply:
but even that doesn't work that way. hence using count instead.